lib/remap-functions/src/del.rs

```diff
@@ -9,47 +9,54 @@ impl Function for Del {
     }
 
     fn parameters(&self) -> &'static [Parameter] {
-        generate_param_list! {
-            accepts = |_| true,
-            required = false,
-            keywords = [
-                "1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13", "14", "15", "16",
-            ],
-        }
+        &[Parameter {
+            keyword: "field",
+            accepts: |_| true,
+            required: true,
+        }]
     }
 
     fn compile(&self, mut arguments: ArgumentList) -> Result<Box<dyn Expression>> {
-        let mut paths = vec![];
-        paths.push(arguments.required_path("1")?);
-
-        for i in 2..=16 {
-            if let Some(path) = arguments.optional_path(&format!("{}", i))? {
-                paths.push(path)
-            }
-        }
+        let field = arguments.required_path("field")?;
 
-        Ok(Box::new(DelFn { paths }))
+        Ok(Box::new(DelFn { field }))
     }
 }
 
 #[derive(Debug, Clone)]
 pub struct DelFn {
-    paths: Vec<Path>,
+    field: Path,
+}
+
+impl DelFn {
+    #[cfg(test)]
+    fn new(field: Path) -> Self {
+        Self { field }
+    }
 }
 
 impl Expression for DelFn {
     fn execute(&self, _: &mut state::Program, object: &mut dyn Object) -> Result<Value> {
-        self.paths
-            .iter()
-            .try_for_each(|path| object.remove(path.as_ref(), false))?;
-
-        Ok(Value::Null)
+        match object.remove_and_get(self.field.as_ref(), false) {
```

Comment:
We should probably just update `remove` to always return the removed value.

You can always ignore that value if it's not of use.